internal/provider/resource_offer.go

```diff
@@ -2,174 +2,261 @@ package provider
+func (o *offerResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	// Prevent panic if the provider has not been configured.
+	if o.client == nil {
+		addClientNotConfiguredError(&resp.Diagnostics, "offer", "delete")
+		return
+	}
+	var plan offerResourceModel
 
-	var diags diag.Diagnostics
+	// Get the Terraform state from the request into the plan
+	resp.Diagnostics.Append(req.State.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
 
-	result, err := client.Offers.ReadOffer(&juju.ReadOfferInput{
-		OfferURL: d.Id(),
+	err := o.client.Offers.DestroyOffer(&juju.DestroyOfferInput{
+		OfferURL: plan.URL.ValueString(),
 	})
 	if err != nil {
-		return handleOfferNotFoundError(err, d)
+		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to delete offer, got error: %s", err))
+		return
 	}
+	tflog.Trace(ctx, fmt.Sprintf("delete offer resource %q", plan.URL))
+}
 
-	if err = d.Set("model", result.ModelName); err != nil {
-		return diag.FromErr(err)
-	}
-	if err = d.Set("name", result.Name); err != nil {
-		return diag.FromErr(err)
-	}
-	if err = d.Set("application_name", result.ApplicationName); err != nil {
-		return diag.FromErr(err)
-	}
-	if err = d.Set("endpoint", result.Endpoint); err != nil {
-		return diag.FromErr(err)
+func (o *offerResource) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
 	}
-	if err = d.Set("url", result.OfferURL); err != nil {
-		return diag.FromErr(err)
+
+	client, ok := req.ProviderData.(*juju.Client)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *juju.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+		return
 	}
-	d.SetId(result.OfferURL)
 
-	return diags
+	o.client = client
 }
 
-func resourceOfferDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-	client := meta.(*juju.Client)
+func (o *offerResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
+}
 
-	var diags diag.Diagnostics
+func isOfferNotFound(err error) bool {
+	return strings.Contains(err.Error(), "expected to find one result for url")
+}
 
-	err := client.Offers.DestroyOffer(&juju.DestroyOfferInput{
-		OfferURL: d.Get("url").(string),
-	})
-	if err != nil {
-		return diag.FromErr(err)
+func handleOfferNotFoundError(ctx context.Context, err error, st *tfsdk.State) diag.Diagnostics {
+	if isOfferNotFound(err) {
+		// Offer manually removed
+		st.RemoveResource(ctx)
+		return diag.Diagnostics{}
 	}
 
-	d.SetId("")
-
+	var diags diag.Diagnostics
+	diags.AddError("Not Found", err.Error())
```

Comment:
We exited on line 256, thus do not have a 'not found' error. Please update the summary to "Client error".